src/main/java/org/ladocuploader/app/interceptors/DataRequiredInterceptor.java

```diff
@@ -1,83 +1,150 @@
 package org.ladocuploader.app.interceptors;
 
 import formflow.library.FormFlowController;
+import formflow.library.data.Submission;
 import formflow.library.data.SubmissionRepositoryService;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-import jakarta.servlet.http.HttpSession;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Component;
 import org.springframework.util.AntPathMatcher;
 import org.springframework.web.servlet.HandlerInterceptor;
 
+import java.io.IOException;
 import java.util.Map;
 import java.util.UUID;
 
 @Component
 @Slf4j
 public class DataRequiredInterceptor implements HandlerInterceptor {
-    public static final String PATH_FORMAT = "/flow/{flow}/{screen}";
-    private final SubmissionRepositoryService submissionRepositoryService;
-    private static final Map<String, String> REQUIRED_DATA = Map.ofEntries(
-            // Step 1
-            Map.entry("howToAddDocuments", "firstName"),
-            Map.entry("uploadDocuments", "firstName"),
-            Map.entry("docSubmitConfirmation", "firstName"),
-            Map.entry("finalConfirmation", "firstName")
-    );
-
-    public DataRequiredInterceptor(SubmissionRepositoryService submissionRepositoryService) {
-        this.submissionRepositoryService = submissionRepositoryService;
-    }
+  public static final String PATH_FORMAT = "/flow/{flow}/{screen}";
+  private final SubmissionRepositoryService submissionRepositoryService;
+  private static final Map<String, String> REQUIRED_DATA = Map.ofEntries(
+      Map.entry("laDocUpload", "firstName"),
+      Map.entry("laDigitalAssister", "languageRead")
+  );
+
+  private static final Map<String, String> BEGINNING_PAGES = Map.ofEntries(
+      Map.entry("laDocUpload", "clientInfo"),
+      Map.entry("laDigitalAssister", "languagePreference")
+  );
+
+  public DataRequiredInterceptor(SubmissionRepositoryService submissionRepositoryService) {
+    this.submissionRepositoryService = submissionRepositoryService;
+  }
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+    try {
+      var parsedUrl = new AntPathMatcher().extractUriTemplateVariables(PATH_FORMAT, request.getRequestURI());
+      var screenName = parsedUrl.get("screen");
+      if (BEGINNING_PAGES.containsValue(screenName) || "GET".equals(request.getMethod())) {
+        // We dont have values to lose on beginning pages and we're not handling GETs right now
+        return true;
+      }
+
+      var session = request.getSession(false);
+      if (session == null) {
+        return handleMissingSession(request, response);
+      }
+      
+      Map<String, UUID> submissionMap = (Map) session.getAttribute(FormFlowController.SUBMISSION_MAP_NAME);
```

Comment:
flow name to session id - is this handled correctly with recovering the submissionId?